src/go/rpk/pkg/cli/cmd/plugin/plugin.go

```diff
@@ -0,0 +1,297 @@
+			}
+
+			m, err := getManifest()
+			out.MaybeDieErr(err)
+
+			tw := out.NewTable("NAME", "DESCRIPTION", "MESSAGE")
+			defer tw.Flush()
+			for _, entry := range m.Plugins {
+				name := entry.Name
+				_, entrySha, _ := entry.PathShaForUser()
+
+				var message string
+
+				p, exists := installed.Find(name)
+				if exists {
+					name = "*" + name
+
+					sha, err := plugin.Sha256Path(fs, p.Path)
+					if err != nil {
+						message = fmt.Sprintf("unable to calculate local binary sha256: %v", err)
+					} else if sha != entrySha {
+						message = "local binary sha256 differs from manifest sha256"
+					}
+				}
+
+				tw.Print(name, entry.Description, message)
+			}
+		},
+	}
+
+	cmd.Flags().BoolVarP(&local, "local", "l", false, "list locally installed plugins and shadowed plugins")
+
+	return cmd
+}
+
+func newInstallCommand(fs afero.Fs) *cobra.Command {
+	var dir string
+	var update bool
+	cmd := &cobra.Command{
+		Use:     "install [PLUGIN]",
+		Aliases: []string{"download"},
+		Short:   "Install an rpk plugin.",
+		Long: `Install an rpk plugin.
+
+An rpk plugin must be saved in a directory that is in your $PATH. By default,
+this command installs plugins to the first directory in your $PATH. This can
+be overridden by specifying the --bin-dir flag.
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(_ *cobra.Command, args []string) {
+			name := args[0]
+
+			fmt.Printf("Searching plugin manifest for %q...\n", name)
+			m, err := getManifest()
+			out.MaybeDieErr(err)
+
+			p, err := m.FindEntry(name)
+			out.MaybeDieErr(err)
+
+			_, remoteSha, err := p.PathShaForUser()
+			out.MaybeDieErr(err)
+
+			var found bool
+			installed := plugin.ListPlugins(fs, plugin.UserPaths())
+			for _, p := range installed {
+				if name == p.Name() {
+					sha, err := plugin.Sha256Path(fs, p.Path)
+					out.MaybeDieErr(err)
+
+					if sha != remoteSha {
+						msg := fmt.Sprintf(`Plugin %q is already installed, but is different from the remote plugin!
+
+ Local sha256: %s
+Remote sha256: %s
+
+`, name, sha, remoteSha)
+						if !update {
+							out.Exit(msg + "--update was not requested, exiting!")
+						}
+						fmt.Println(msg + "Downloading and validating updated plugin...")
+						found = true
+
+					} else {
+						out.Exit("Plugin %q is already installed and up to date!", name)
+					}
+				}
+			}
+			if !found {
+				fmt.Println("Found! Downloading and validating plugin...")
+			}
+
+			body, err := p.DownloadForUser(urlBase)
+			out.MaybeDieErr(err)
+
+			fmt.Println("Downloaded! Writing plugin to disk...")
+			dst, err := plugin.WriteBinary(fs, p.Name, dir, body, p.HelpAutoComplete)
+			out.MaybeDieErr(err)
+
+			fmt.Printf("Success! Plugin %q has been saved to %q and is now ready to use!\n", p.Name, dst)
+
+			if p.HelpAutoComplete {
+				fmt.Printf(`
+This plugin supports autocompletion through rpk.
+
+If you enable rpk autocompletion, start a new terminal to tab complete your new
+command %q!
+`, p.Name)
+			} else {
+				fmt.Printf(`
```

Comment:
Why not use `logrus` like the rest of the code?

Reply:
This is the stylized output for the console, logrus wouldn't really be relevant here. We use logrus for debug logs.

Related, though, I don't think logrus is the best choice for a CLI. logrus provides structured output that makes it much easier to process server logs for analysis / dashboards. CLIs are run ad hoc and quickly, the structured output aspect is not overkill and is more difficult for a human to parse. It may be better through some gradual refactoring to convert to something nicer for a CLI.

Reply:
I think this makes sense. Nevertheless, some rpk commands will be used in automated processes that might wanna get timestamped logs. Then again, many CI services add the timestamp themselves so I'm on the fence.

Reply:
This same thought would need to be applied to every other command that has any output -- and we don't add timestamps to all of those.

Also, I kinda have the same thought on automated processes -- every system I've worked with has had the process wrapped in something that timestamps logs, so any process that logged with timestamps then had two timestamps in the output... I eventually removed the duplication :)